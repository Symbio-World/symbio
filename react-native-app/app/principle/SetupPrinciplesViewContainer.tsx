import React from 'react'
import { SetupPrinciplesView } from './SetupPrinciplesView'
import { useAuth } from '../auth'
import { savePrinciples } from './savePrinciples'
import { getAllPrinciples } from './getAllPrinciples'

const principles = getAllPrinciples()

export type OnSave = () => void
type Props = {
  onSave?: () => void
}

export const SetupPrinciplesViewContainer: React.FC<Props> = ({
  onSave = () => {},
}) => {
  const { user } = useAuth()

  const handleSubmit = (selectedPrinciples: string[]) => {
    if (!user) {
      throw new Error('This should never happen, as user is in the context')
    }
    savePrinciples(user.id, selectedPrinciples)
    onSave()
  }

  return <SetupPrinciplesView principles={principles} onSubmit={handleSubmit} />
}
