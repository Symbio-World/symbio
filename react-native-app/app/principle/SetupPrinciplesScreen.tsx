import * as React from 'react'
import { StackNavigationProp } from '@react-navigation/stack'
import { SetupPrinciplesViewContainer } from './SetupPrinciplesViewContainer'
import { RootStackParamList } from '../Navigation'

type Props = {
  navigation: StackNavigationProp<RootStackParamList, 'SetupPrinciplesScreen'>
}
export const SetupPrinciplesScreen: React.FC<Props> = ({ navigation }) => {
  const handleSave = () => {
    navigation.navigate('ScanBarcodeScreen')
  }

  return <SetupPrinciplesViewContainer onSave={handleSave} />
}
