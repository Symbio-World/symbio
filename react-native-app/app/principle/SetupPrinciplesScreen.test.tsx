import 'react-native'
import * as React from 'react'
import { render, fireEvent } from 'react-native-testing-library'
import { SetupPrinciplesScreen } from './SetupPrinciplesScreen'
import { SetupPrinciplesViewContainer } from './SetupPrinciplesViewContainer'

jest.mock('@react-navigation/native')

describe('SetupPrinciplesScreen', () => {
  const navigation: any = {
    navigate: jest.fn(),
  }

  it('renders correctly', () => {
    const { toJSON } = render(<SetupPrinciplesScreen navigation={navigation} />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('navigates out onSave', () => {
    const { getByType } = render(
      <SetupPrinciplesScreen navigation={navigation} />,
    )
    fireEvent(getByType(SetupPrinciplesViewContainer), 'onSave')
    expect(navigation.navigate).toHaveBeenCalledWith('ScanBarcodeScreen')
  })
})
