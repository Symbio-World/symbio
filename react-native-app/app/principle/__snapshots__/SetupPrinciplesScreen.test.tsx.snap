// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetupPrinciplesScreen renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "margin": 20,
      },
    ]
  }
>
  <RCTScrollView>
    <View>
      <View>
        <Text
          style={
            Array [
              Object {
                "fontSize": 36,
              },
              Object {
                "fontWeight": "bold",
              },
              Object {
                "marginVertical": 16,
              },
            ]
          }
        >
          ðŸ‘‹Hey There
        </Text>
        <Text
          style={
            Array [
              Object {
                "marginVertical": 16,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Should I Buy
          </Text>
           lets you understand if the products you find in the supermarket fit your values.
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontWeight": "bold",
              },
            ]
          }
        >
          What is important to you?
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              Object {
                "flexWrap": "wrap",
              },
              Object {
                "marginVertical": 16,
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#48bb78",
                "borderRadius": 9999,
                "borderWidth": 2,
                "margin": 12,
                "opacity": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#48bb78",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Vegan
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#48bb78",
                "borderRadius": 9999,
                "borderWidth": 2,
                "margin": 12,
                "opacity": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#48bb78",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Vegeterian
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#48bb78",
                "borderRadius": 9999,
                "borderWidth": 2,
                "margin": 12,
                "opacity": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#48bb78",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Pescatarian
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#48bb78",
                "borderRadius": 9999,
                "borderWidth": 2,
                "margin": 12,
                "opacity": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#48bb78",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Gluten free
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#48bb78",
                "borderRadius": 9999,
                "borderWidth": 2,
                "margin": 12,
                "opacity": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#48bb78",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Nut free
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#48bb78",
                "borderRadius": 9999,
                "borderWidth": 2,
                "margin": 12,
                "opacity": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#48bb78",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Lactose free
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#48bb78",
                "borderRadius": 9999,
                "borderWidth": 2,
                "margin": 12,
                "opacity": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#48bb78",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Halal
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#48bb78",
                "borderRadius": 9999,
                "borderWidth": 2,
                "margin": 12,
                "opacity": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#48bb78",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Organic
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#48bb78",
                "borderRadius": 9999,
                "borderWidth": 2,
                "margin": 12,
                "opacity": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#48bb78",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Eco friendly
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#48bb78",
                "borderRadius": 9999,
                "borderWidth": 2,
                "margin": 12,
                "opacity": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#48bb78",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Local products
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#48bb78",
                "borderRadius": 9999,
                "borderWidth": 2,
                "margin": 12,
                "opacity": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#48bb78",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              No sugar
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#48bb78",
                "borderRadius": 9999,
                "borderWidth": 2,
                "margin": 12,
                "opacity": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#48bb78",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              No sodium
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#48bb78",
                "borderRadius": 9999,
                "borderWidth": 2,
                "margin": 12,
                "opacity": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#48bb78",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Fair trade
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#48bb78",
                "borderRadius": 9999,
                "borderWidth": 2,
                "margin": 12,
                "opacity": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#48bb78",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Package free
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#48bb78",
                "borderRadius": 9999,
                "borderWidth": 2,
                "margin": 12,
                "opacity": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#48bb78",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Recyclable
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#48bb78",
                "borderRadius": 9999,
                "borderWidth": 2,
                "margin": 12,
                "opacity": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#48bb78",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Compostable
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 96,
            },
          ]
        }
      />
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "bottom": 0,
        },
        Object {
          "width": "100%",
        },
        Object {
          "backgroundColor": "#fff",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#B35340",
          "borderRadius": 4,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "fontSize": 24,
            },
            Object {
              "color": "#fff",
            },
            Array [
              Object {
                "color": "#fff",
              },
            ],
          ]
        }
      >
        Get Started
      </Text>
    </View>
  </View>
</View>
`;
